native/src/main/java/io/ballerina/stdlib/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,271 @@
+     }
+ 
+     /**
+      * Creates a provider specific json array
+      *
+      * @return new array
+      */
+     @Override
+     public Object createArray() {
+         return ValueCreator.createArrayValue(PredefinedTypes.TYPE_JSON_ARRAY);
+     }
+ 
+     /**
+      * Creates a provider specific json object
+      *
+      * @return new object
+      */
+     @Override
+     public Object createMap() {
+         return ValueCreator.createMapValue();
+     }
+ 
+     public boolean isArray(Object obj) {
+         return (obj instanceof BArray);
+     }
+ 
+     public Object getArrayIndex(Object obj, int idx) {
+         return toJsonArray(obj).get(idx);
+     }
+ 
+     public void setArrayIndex(Object array, int index, Object newValue) {
+         if (!isArray(array)) {
+             throw new UnsupportedOperationException();
+         }
+         BArray arrayNode = toJsonArray(array);
+         if (index == arrayNode.size()) {
+             arrayNode.append(createJsonElement(newValue));
+         } else {
+             arrayNode.add(index, createJsonElement(newValue));
+         }
+     }
+ 
+     public Object getMapValue(Object obj, String key) {
+         if (!isMap(obj)) {
+             throw new UnsupportedOperationException();
+         }
+         BMap jsonObject = toJsonObject(obj);
+         if (!jsonObject.containsKey(StringUtils.fromString(key))) {
+             return UNDEFINED;
+         } else {
+             return unwrap(jsonObject.get(StringUtils.fromString(key)));
+         }
+     }
+ 
+     public void setProperty(Object obj, Object key, Object value) {
+         if (isMap(obj)) {
+             toJsonObject(obj).put(key.toString(), createJsonElement(value));
+         } else {
+             BArray array = toJsonArray(obj);
+             long index;
+             if (key != null) {
+                 index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+             } else {
+                 index = array.size();
+             }
+ 
+             if (index == array.size()) {
+                 array.append(createJsonElement(value));
+             } else {
+                 array.add(index, createJsonElement(value));
+             }
+         }
+     }
+ 
+     public void removeProperty(Object obj, Object key) {
+         if (isMap(obj)) {
+             toJsonObject(obj).remove(key.toString());
+         } else {
+             BArray array = toJsonArray(obj);
+             int index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+             array.shift(index);
+         }
+     }
+ 
+     public boolean isMap(Object obj) {
+         return (obj instanceof BMap);
+     }
+ 
+     public Collection<String> getPropertyKeys(Object obj) {
+         List<String> keys = new ArrayList<>();
+         for (Object entry : toJsonObject(obj).getKeys()) {
+             keys.add(entry.toString());
+         }
+         return keys;
```

Comment:
Can't we make this more compact in this way, 
```suggestion
         return Arrays.stream(toJsonObject(obj).getKeys())
                .map(Object::toString)
                .collect(Collectors.toList());
```